crates/sui-indexer-alt-framework/src/lib.rs

```diff
@@ -228,24 +246,23 @@ impl<S: Store> Indexer<S> {
             return Ok(());
         };
 
-        // If `first_checkpoint` does not violate the consistency check, concurrent pipelines will
-        // prefer to resume from the `first_checkpoint` if configured.
         let next_checkpoint = match (watermark, self.first_checkpoint) {
             (Some(watermark), Some(first_checkpoint)) => {
-                // Setting `skip_watermark` allows concurrent pipelines to not be considered in the
-                // consistency check. The indexer will still fail to start if `first_checkpoint`
-                // fails for a sequential pipeline in the indexer.
-                if !self.skip_watermark {
+                // If this is not a tasked indexer, we're dealing with main pipelines. Check that
+                // the `--first-checkpoint` is not greater than this pipeline's committer watermark,
+                // as this would cause the pipeline to stall forever in wait of a checkpoint that
+                // will never be ingested.
```

Comment:
IIUC, this works differently to how I expected, which was that we treat `--first-checkpoint` the same for both kinds of pipeline (tasked vs not), in that we listen to it only if there is not an existing watermark, and otherwise ignore it (perhaps issuing a warning).

This implementation seems like it will not ignore a `--first-checkpoint` for main pipelines, even if a watermark task exists.